File: Programming/Java/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.java
public class RemoveDuplicatesFromArray {
	public int removeDuplicates(int[] nums) {		
		int i=1;
		if(nums.length==0) {
			i=0;
		}
		else {
		    for(int j=1;j<nums.length;j++){
			    if(nums[j]!=nums[j-1]){         
				    nums[i]=nums[j];
				    i++;
			     }
		     }
		}
	     return i;
	}
}

File: Programming/Golang/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("This program will remove duplicate elements from a sorted array!")
	fmt.Println(removeDuplicates([]int{1}))
	fmt.Println(removeDuplicates([]int{1, 1, 2}))
	fmt.Println(removeDuplicates([]int{0,0,1,1,1,2,2,3,3,4}))
	fmt.Println(removeDuplicates([]int{2, 2, 2, 2, 2}))
	fmt.Println(removeDuplicates([]int{1, 2, 2, 3, 4, 4, 4, 5, 5}))
}


func removeDuplicates(nums []int) []int {
    m := make(map[int]int)

    for i := range nums {
        if _,k := m[nums[i]] ; k {
            m[nums[i]]+=1
        } else {
            m[nums[i]]=1
        }
    }
   
    for i :=0 ; i<len(nums);i++ {
        nums = append(nums[:i+1] , nums[i+m[nums[i]]: ]...)
    }
    return nums
}